src/shogun/base/SGObject.h

```diff
@@ -390,9 +393,34 @@ class CSGObject
 		put(Tag<T*>(name), value);
 	}
 
+    /** Typed setter for an enum mapped with a string, identified by a name.
+     *
+     * @param name name of the parameter
+     * @param value value of the parameter
+	 */
+	template <class T>
+	void put(const std::string& name, const std::string& value) noexcept(false)
+	{
+		if (m_enum_to_string_map.find(get_name()) == m_enum_to_string_map.end())
```

Comment:
Would remove and just keep the below error. Same as above,  only confuses users, it’s more an internal thing